### PR TITLE
Render still API

### PR DESCRIPTION
**Pull request by JonnyBurger**
- [x] Ensure output path is valid
- [x] Throw if composition ID does not exists
- [x] CLI reference
- [x] `<Still>` component

### CODE REVIEW

File: packages/cli/src/index.tsx [hunk 1]
```diff
@@ -22,6 +23,8 @@ export const cli = async () => {
 		await previewCommand();
```
Comment (JonnyBurger) [inline packages/cli/src/index.tsx:23]: Should add to printHelp()

File: packages/cli/src/parse-command-line.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export type CommandLineOptions = {
 	log: string;
 	help: boolean;
 	port: number;
+	['frame']: string | number;
```
Comment (JonnyBurger) [inline packages/cli/src/parse-command-line.ts]: Why with square brackets?

File: packages/cli/src/parse-command-line.ts [hunk 3]
```diff
@@ -77,6 +78,10 @@ export const parseCommandLine = () => {
 		Internals.setFrameRangeFromCli(parsedCli.frames);
 	}
 
+	if (parsedCli.frame) {
+		Internals.setStillFrame(Number(parsedCli.frame));
```
Comment (JonnyBurger) [inline packages/cli/src/parse-command-line.ts:96]: Throw error if not using `still` command?

File: packages/core/src/test/validate-image-format.test.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import {validateImageFormat} from '../validation/validate-image-format';
+
+test('"none" is not a valid image format', () => {
```
Comment (JonnyBurger) [inline packages/core/src/test/validate-image-format.test.ts:3]: Rename variable to make it clear that "none" is not accepted?

File: packages/core/src/validation/validate-quality.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	if (!Number.isFinite(q)) {
+		throw new RangeError(`Quality must be a finite number, but is ${q}`);
+	}
+
+	if (Number.isNaN(q)) {
+		throw new RangeError(`Quality is NaN, but must be a real number`);
+	}
+
+	if (q > 100 || q < 0) {
+		throw new RangeError('Quality option must be between 1 and 100.');
```
Comment (JonnyBurger) [inline packages/core/src/validation/validate-quality.ts]: between 0 and 100

File: packages/docs/docs/render-still.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
+---
+id: render-still
+title: renderStill()
+---
+
+_Part of the `@remotion/renderer` package. Available from v2.3._
+
+Renders a single frame to an image.
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: and writes it to an output location

File: packages/docs/docs/render-still.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+If you want to render the full image sequence, use [renderFrames()](/docs/render-frames) instead.
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: render a full image sequence (and possibly encode it to a video later)

File: packages/docs/docs/render-still.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+if (!composition) {
+  throw new Error(`No composition with the ID ${compositionId} found`)
+}
+
+// ---cut---
+
+await renderStill({
+  composition,
+  webpackBundle: bundled,
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: bundled -> bundleLocation

File: packages/docs/docs/render-still.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+### `webpackBundle`
+
+The absolute location of your webpack bundle. Use [`bundle()`](/docs/bundle) to bundle your project programmatically.
+
+### `output`
+
+An absolute path to where the frame should be rendered to.
+
+### `inputProps?`
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md:69]: _optional_

File: packages/docs/docs/render-still.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+[Custom props which will be passed to the component.](/docs/parametrized-rendering) Useful for rendering videos with dynamic content. Can be an object of any shape.
+
+### `frame?`
+
+_optional - default: 0_
+
+Which frame should be rendered based on it's number.
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: based on its

File: packages/docs/docs/render-still.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+### `imageFormat?`
+
+_optional - default: "png"_
+
+Which output format the image should have, either `png` or `jpeg`.
+
+### `quality?`
+
+_optional - default: 80_
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: default undefined

File: packages/docs/docs/render-still.md [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+_optional - default `null`_
+
+An already open Puppeteer [`Browser`](https://pptr.dev/#?product=Puppeteer&version=main&show=api-class-browser) instance. Reusing a browser across multiple function calls can speed up the rendering process. You are responsible for opening and closing the browser yourself. If you don't specify this option, a new browser will be opened and closed at the end.
+
+### `envVariables?`
+
+_optional - default `{}`_
+
+An object containing key-value pairs of environment variables which will be injected into your Remotion projected and which can be accessed by reading the global `process.env` object.
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: into your Remotion project

File: packages/docs/docs/render-still.md [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+
+### `dumpBrowserLogs?`
+
+_optional - default `false`_
+
+A boolean value deciding whether Puppeteer logs should be printed to the console.
```
Comment (JonnyBurger) [inline packages/docs/docs/render-still.md]: useful for debugging only

File: packages/docs/docs/render.md [hunk 14]
```diff
@@ -18,6 +18,14 @@ npx remotion render src/index.tsx HelloWorld out.mp4
...
 
+## More ways to render a video
+
+Remotion has a full-featured server-side rendering API. Read more about it on the [server-side rendering API](/docs/ssr).
+
+You can also render a [video using a Github Action.](/docs/ssr#render-using-github-actions)
+
+We are also working on a Serverless solution for rendering videos, [which is in private beta](/docs/ssr#rendering-a-video-using-serverless).
+
 ## See also
```
Comment (JonnyBurger) [inline packages/docs/docs/render.md:47]: Mention rendering a still here

File: packages/cli/src/still.ts [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+		inputProps,
+		onError: (err: Error) => {
+			Log.error();
+			Log.error('The following error occured when rendering the still:');
+
+			handleCommonError(err);
+
+			process.exit(1);
+		},
+		puppeteerInstance: openedBrowser,
```
Comment (JonnyBurger) [inline packages/cli/src/still.ts:123]: should respect overwrite flag

File: packages/renderer/src/render-still.ts [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+		'in the `config` object of `renderStill()`'
+	);
+	Internals.validateDurationInFrames(
+		composition.durationInFrames,
+		'in the `config` object passed to `renderStill()`'
+	);
+	Internals.validateImageFormat(imageFormat);
+	Internals.validateFrame(frame, composition.durationInFrames);
+
+	output = path.resolve(process.cwd(), output);
```
Comment (JonnyBurger) [inline packages/renderer/src/render-still.ts:68]: Log verbose output? what if output was forgotten to be passed?

File: packages/cli/src/print-help.ts [hunk 17]
```diff
@@ -42,6 +42,21 @@ export const printHelp = () => {
 		['--port', 'Custom port to use for the HTTP server'],
 		['--env-file', 'Specify a location for a dotenv file'],
 	]);
+	Log.info('remotion still <index-file.ts> <comp-name> <still.mp4>');
```
Comment (Iamshankhadeep) [inline packages/cli/src/print-help.ts]: need to change to 'still.png'

**Review by Iamshankhadeep — CHANGES_REQUESTED** [review]
While testing I found a bug it still renders .mp4 videos and each video is 10 sec long.

**Pull request merged**